plone/app/contenttypes/behaviors/collection.py

```diff
@@ -16,23 +16,6 @@
 from zope.interface import implementer
 from zope.interface import provider
 from zope.schema.interfaces import IVocabularyFactory
-from zope.schema.vocabulary import SimpleTerm
-from zope.schema.vocabulary import SimpleVocabulary
-
-
-@implementer(IVocabularyFactory)
-class MetaDataFieldsVocabulary(object):
-
-    def __call__(self, context):
-        cat = getToolByName(context, 'portal_catalog')
-        items = [
-            SimpleTerm(column, column, column)
-            for column in cat.schema()
-        ]
-        return SimpleVocabulary(items)
-
-
-MetaDataFieldsVocabularyFactory = MetaDataFieldsVocabulary()
```

Comment:
I like the work you did on the vocabulary and it definetely makes sense to move it to a vocabulary module.
Given I think we want to have in also in Plone 5.2, I would think about a strategy to not make it a breaking change.
I see two possibility:
1. leave the vocabulary here 
2. at least add a `zope.deferredimport`:
```
zope.deferredimport.deprecated(
    "Import from plone.app.contenttypes.vocabulary instead (this compatibility layer will be removed in Plone 6)",
    MetaDataFieldsVocabulary='plone.app.contenttypes:vocabulary.MetaDataFieldsVocabulary',
)
```
The same for `MetaDataFieldsVocabularyFactory`.
I see the chance of that being used by third party code is low, I know. But similar moves in the past already caused troubles.

What do you think?

Reply:
I did the deprecated imports. Thanks for the suggestion.